feat(@angular/cli) override suite in the protractor config

aniruddhadas9 opened the pull request:
feat(@angular/cli) override suite in the protractor config

resolves: 807
Override suite in the protractor config.
Can send in multiple suite by comma seperated values (ng e2e --suite=suite1.ts, suite2.ts).

Issue link
 github.com/angular/angular-cli/issues/807
 github.com/angular/angular-cli/pull/3551

aniruddhadas9 commented:
@filipesilva started writing e2e and will append them to this pull request.

dallasacook commented:
@filipesilva Have the changes met your requests?

dallasacook commented:
@aniruddhadas9 can you take a look at why the CI checks are failing for this? I would love to see this feature merged.

aniruddhadas9 commented:
@dallasacook sure, I will look into it and fix this build issue.

aniruddhadas9 commented:
@filipesilva, I have updated code and test case. I have manually tested all possible scenario to make sure everything working fine. The Travis job is failing due to commit message, I think its because of the merge commit happened when I updated code from  base and merged my code. Please let me know if anything else need to be taken care off.

filipesilva commented on tests/e2e/assets/1.0.0-proj/protractor.conf.js:
```diff
@@ -17,15 +17,16 @@ exports.config = {
   jasmineNodeOpts: {
```
Can you remove the changes to this file? It's true that they are just harmless whitespace, but they are also unnecessary so we shouldn't change it without a good reason.

filipesilva commented on tests/e2e/tests/basic/e2e.ts:
```diff
@@ -34,6 +34,21 @@ export default function () {
     .then(() => ng('e2e', '--specs', './e2e/renamed-app.e2e-spec.ts',
       '--specs', './e2e/another-app.e2e-spec.ts'))
+    // Suites block need to be added in the protractor.conf.js file to test suites
+    .then(() => replaceInFile('protractor.conf.js', `allScriptsTimeout: 11000,`,
```
I think this is a good test. I tried to do this without the your PR and e2e broke with `Error: Unknown test suite: app`.

aniruddhadas9 replied:
Thank you :). Yap, I also tried same and some other combination and thought to add suites and then test it.

aniruddhadas9 commented:
@filipesilva Thanks for reviewing :) . Now merged all of them to one commit and removed formatting related changes in the `protractor.conf.js` file. I think everything will go fine now.

aniruddhadas9 commented:
@filipesilva its failing due to unable to connect to server now.


```

Cloning into 'angular/angular-cli'...
fatal: unable to access 'https://github.com/angular/angular-cli.git/': Failed to connect to github.com port 443: Connection timed out
The command "eval git clone --depth=50 https://github.com/angular/angular-cli.git angular/angular-cli " failed. Retrying, 2 of 3.
Cloning into 'angular/angular-cli'...
fatal: unable to access 'https://github.com/angular/angular-cli.git/': Failed to connect to github.com port 443: Connection timed out
The command "eval git clone --depth=50 https://github.com/angular/angular-cli.git angular/angular-cli " failed. Retrying, 3 of 3.
Cloning into 'angular/angular-cli'...
fatal: unable to access 'https://github.com/angular/angular-cli.git/': Failed to connect to github.com port 443: Connection timed out
The command "eval git clone --depth=50 https://github.com/angular/angular-cli.git angular/angular-cli " failed 3 times.
Failed to clone from GitHub.
Checking GitHub status (https://status.github.com/api/last-message.json):
good
Everything operating normally.
2017-11-28T00:03:47Z

```

filipesilva commented:
@aniruddhadas9 thanks for taking the time to implement this feature! It should be on the next minor version, `1.7` (we don't publish features on patch version `1.6.x`).
